src/main/java/seedu/duke/commands/AddCommand.java

```
@@ -1,36 +1,84 @@
 
-    public static void addEntry(String description, String category, double amount){
-        Entry newEntry;
+    private Entry entryObj;
+
+    /**
+     * AddCommand constructor which initialises entryObj to be added
+     *
+     * @param description Description of the entry
+     * @param category    Category which entry belongs to
+     * @param amount      Price of entry
+     */
+    public AddCommand(String description, String category, double amount) {
         switch(category){
-        case CATEGORY_FOOD:
-            newEntry = new Food(description, amount);
+        case CATEGORY_CLOTHING:
+            this.entryObj = new Entry(description, amount, Category.CLOTHING);
+            break;
+
+        case CATEGORY_ENTERTAINMENT:
+            this.entryObj = new Entry(description, amount, Category.ENTERTAINMENT);
             break;
 
-        case CATEGORY_OTHER:
-            newEntry = new Other(description, amount);
+        case CATEGORY_FOOD:
+            this.entryObj = new Entry(description, amount, Category.FOOD);
             break;
 
         case CATEGORY_INCOME:
-            newEntry = new Income(description, amount);
+            this.entryObj = new Entry(description, amount, Category.INCOME);
+            break;
+
+        case CATEGORY_MEDICAL:
+            this.entryObj = new Entry(description, amount, Category.MEDICAL);
+            break;
+
+        case CATEGORY_OTHERS:
+            this.entryObj = new Entry(description, amount, Category.OTHERS);
+            break;
+
+        case CATEGORY_PERSONAL:
+            this.entryObj = new Entry(description, amount, Category.PERSONAL);
+            break;
+
+        case CATEGORY_TRANSPORTATION:
+            this.entryObj = new Entry(description, amount, Category.TRANSPORTATION);
+            break;
+
+        case CATEGORY_UTILITIES:
+            this.entryObj = new Entry(description, amount, Category.UTILITIES);
             break;
 
         default:
             return;
```

Instead of returning, I think an exception should be thrown if an invalid category is entered. Furthermore, this will prevent the adding of a `null` object to `EntryLog`